### PR TITLE
docs: added docker instructions to the exercises

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Current Behavior

This PR addresses the challenges users may encounter when conducting the learning exercises in a local development environment.  An example of an issue is #3521 , more examples can be found in the help thread.

## Proposed Changes

The PR adds instructions on using a Docker container for the exercises featured in the main `README.md`. The user may still proceed with a local install should they choose to.  The benefit is a better learning experience for the user and less time spent by the Ockam team on answering questions about a user's local environment, which can be difficult and time-consuming to debug.

There is a price to pay with this setup. The first compile session will take a few minutes due to updates to crates.io index. 
In my opinion, the tradeoff of less time spent in helping users debug local environment issues makes this worth it for all parties.

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.  Please see --> https://github.com/build-trust/ockam-contributors/pull/111

### CODE REVIEW

**Comment by karl-cardenas-coding** [comment]
@mrinalwadhwa no rush on this PR. I went through all the exercises just to make sure I didn't miss anything, but I would appreciate a second glance 😄  Feel free to suggest changes or provide feedback.  I have more ideas on how we can improve the learning, but I think just getting out of the debugging users' local environment will be a good win, to begin with.

**Comment by mrinalwadhwa** [comment]
@karl-cardenas-coding thank you for spending time on this!
I'll read through the changes soon and leave suggestions if I have any.
Congratulations on an awesome first contribution 🥳

**Comment by mrinalwadhwa** [comment]
A couple of quick things, if you have time to improve them.

* The Editorconfig check is failing because some lines in the text you've added have trailing whitespace, if you could remove those, that would be great. https://github.com/build-trust/ockam/actions/runs/3123223435/jobs/5065789655

* The Documentation check is also failing. This one checks to make sure that our examples in guides are not broken and are compilable. I suspect this failure is in the check code itself and its assumptions. Ignore this failure, I'll look into it when I have some more time to read the full PR.

Thank you

**Comment by mrinalwadhwa** [comment]
> removing trailing whitespace

Editors/IDEs usually have builtin commands that do this, more helpful information here https://editorconfig.org/

**Comment by karl-cardenas-coding** [comment]
@mrinalwadhwa nice catch, I'll get that fixed tomorrow.

**Comment by karl-cardenas-coding** [comment]
@mrinalwadhwa linting errors addressed ✔️

File: README.md [hunk 1]
```diff
@@ -56,13 +56,33 @@ For the purpose of our first example, we'll create a local channel within one pr
...
+
+    This command may take a few minutes the first time you invoke it:
+
+    ```
+    docker run --rm -it -e HOST_USER_ID=$(id -u) --name ockam-learn ghcr.io/build-trust/ockam-builder:latest bash
+    ```
+    Upon completion, you will be placed inside the `/work` folder of the container. Next, add a text editior for editing files.
+
+    ```
+    apt update && apt install nano
```
Comment (adrianbenavides) [inline README.md]: I didn't write the original guide but I would say that using `touch` to create the files and not specifying any tool to edit those files is intentional so the user can proceed with their preferred tool (everyone has their own editing preferences after all). Any thoughts on this @mrinalwadhwa ?

**Review by adrianbenavides — COMMENTED** [review]
Hi @karl-cardenas-coding, there's been some recent PR's merged into develop which add some minor modifications to the `README.md` file that conflicts with your changes. Could you please add a commit to resolve the conflicts?

Thanks again for your time, your changes look great!

**Comment by karl-cardenas-coding** [comment]
@adrianbenavides I went ahead and rebased the PR.

**Comment by adrianbenavides** [comment]
@mergifyio rebase

**Comment by adrianbenavides** [comment]
I'm closing this PR as we have included a big change related to how we install/use the ockam binary, which makes most of the changes introduced here outdated. Thanks @karl-cardenas-coding for your work on this PR :pray:

**Pull request closed without merging**